pallets/asset-index/src/lib.rs

```diff
@@ -677,6 +717,43 @@ pub mod pallet {
 			share.checked_div(&nav).ok_or_else(|| ArithmeticError::Overflow.into())
 		}
 
+		/// Consolidate and clean deposits while completing withdraw
+		fn do_consolidate_deposits(caller: &T::AccountId, mut amount: T::Balance) -> Result<T::Balance, DispatchError> {
+			<Deposits<T>>::try_mutate_exists(&caller, |maybe_depositing| -> Result<T::Balance, DispatchError> {
+				let mut depositing = maybe_depositing.take().ok_or(<Error<T>>::NoDeposits)?;
+				let mut total_fee: T::Balance = Default::default();
+
+				let mut dim: Option<(T::Balance, T::BlockNumber)> = None;
+				depositing.retain(|(index_tokens, block_number)| {
+					if amount >= *index_tokens {
```

Comment:
This will also be true for `amount == index_tokens == 0` which should is guaranteed to never happen, right? Can we at least add a note here or better: `if amount.is_zero() {return true}`